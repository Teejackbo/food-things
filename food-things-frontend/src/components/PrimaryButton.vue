<template>
  <button v-bind="attributes" :class="type" v-on="listeners"><slot /></button>
</template>

<script>
export default {
  name: 'PrimaryButton',

  props: {
    type: {
      type: String,
      required: false,
    },
  },

  computed: {
    listeners() {
      // eslint-disable-next-line
      return this.$listeners
    },

    attributes() {
      return this.$attrs
    },
  },
}
</script>

<style lang="scss" scoped>
@import '../style/variables';

button {
  background: $grey-darkest;
  border: 0;
  border-radius: 7px;
  box-shadow: 0 1px 8px $shadow;
  color: $grey;
  font-family: $font-heading;
  font-size: 15px;
  font-weight: 600;
  letter-spacing: 0.5px;
  margin-bottom: 1em;
  padding: 0.4em 1em;
  text-shadow: 0 1px 3px $shadow;
  text-transform: uppercase;
  transition: background 0.3s, transform 0.3s, box-shadow 0.3s, text-shadow 0.3s;

  &:hover {
    background: lighten($grey-darkest, 5%);
    box-shadow: 0 4px 12px $shadow-lighter;
    text-shadow: 0 2px 5px $shadow;
    transform: translateY(-2px);
  }
}

.error {
  background: $red-darker;

  &:hover {
    background: lighten($red-darker, 3%);
  }
}

.success {
  background: $primary-darker;

  &:hover {
    background: lighten($primary-darker, 4%);
  }
}

.warning {
  background: $yellow-darker;
  color: $grey-darkest;

  &:hover {
    background: lighten($yellow-darker, 6%);
  }
}
</style>
