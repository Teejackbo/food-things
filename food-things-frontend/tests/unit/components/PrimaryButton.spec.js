import { shallowMount } from '@vue/test-utils'
import PrimaryButton from '@/components/PrimaryButton.vue'

describe('<PrimaryButton />', () => {
  it('Should match snapshot.', () => {
    const wrapper = shallowMount(PrimaryButton)

    expect(wrapper.element).toMatchSnapshot()
  })

  it('Should render a button element with the class of the type prop.', () => {
    const wrapper = shallowMount(PrimaryButton, {
      propsData: {
        type: 'error',
      },
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
