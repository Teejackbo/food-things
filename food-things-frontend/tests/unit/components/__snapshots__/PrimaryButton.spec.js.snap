// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrimaryButton /> Should match snapshot. 1`] = `<button />`;

exports[`<PrimaryButton /> Should render a button element with the class of the type prop. 1`] = `
<button
  class="error"
/>
`;
